sentry-datasource-proxy/src/main/java/io/sentry/dsproxy/SentryQueryExecutionListener.java

```diff
@@ -0,0 +1,60 @@
+package io.sentry.dsproxy;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.HubAdapter;
+import io.sentry.IHub;
+import io.sentry.ISpan;
+import io.sentry.Span;
+import io.sentry.SpanStatus;
+import io.sentry.util.Objects;
+import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
+import java.util.stream.Collectors;
+import net.ttddyy.dsproxy.ExecutionInfo;
+import net.ttddyy.dsproxy.QueryInfo;
+import net.ttddyy.dsproxy.listener.QueryExecutionListener;
+import org.jetbrains.annotations.NotNull;
+
+/** datasource-proxy query execution listener that creates {@link Span}s around database queries. */
+@Open
+public class SentryQueryExecutionListener implements QueryExecutionListener {
+  private final @NotNull IHub hub;
+  private final @NotNull Map<String, ISpan> spans = new WeakHashMap<>();
+
+  public SentryQueryExecutionListener(final @NotNull IHub hub) {
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+  }
+
+  public SentryQueryExecutionListener() {
+    this(HubAdapter.getInstance());
+  }
+
+  @Override
+  public void beforeQuery(
+      final @NotNull ExecutionInfo execInfo, final @NotNull List<QueryInfo> queryInfoList) {
+    final ISpan parent = hub.getSpan();
+    if (parent != null) {
+      final ISpan span = parent.startChild("db.query", resolveSpanDescription(queryInfoList));
```

Comment:
do we have a standard op and name for this?

similar to https://github.com/getsentry/develop/issues/247

how is in the other SDKs @rhcarvalho maybe @ahmedetefy knows more

Reply:
You can look at how Python does this for inspiration. If you can't find it I can help with more detail next week.

Reply:
Replaced with "db" as it seems to be in Python integration